### DISCUSSION TITLE
New icon menu in route planning

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#444] [Q&A] [answered]
New menu with large buttons in route planning for ease of use, similar to the look of the rest of the app.

Let me know if you like:
- Same color on all buttons, for uniform appearance (top)
- Colored waypoint buttons (bottom)

Map menu|Route menu
-|-
<img src="https://github.com/devemux86/cruiser/assets/3484020/fdcbe455-5d5f-4111-a662-60cc3aa03eda"/>|<img src="https://github.com/devemux86/cruiser/assets/3484020/17c81a88-ae9f-4e77-a732-f959307e482f"/>

Map menu|Route menu
-|-
<img src="https://github.com/devemux86/cruiser/assets/3484020/2b51b666-b728-4cdc-95b1-4329aa8fb3e8"/>|<img src="https://github.com/devemux86/cruiser/assets/3484020/485a94d8-9b41-4dc0-879e-aa4faa8c2cd1"/>

**Comment by airald** [comment]
At first glance, coloured buttons can be easily identifiable, but the purposed icons with same colour are very different and there are no doubt of its meaning, and visually the app is better in my oppinion.

**Comment by devemux86** [comment]
We use icons from the [Material icons](https://fonts.google.com/icons?icon.set=Material+Icons&icon.style=Filled) collection.

If you think we could use a better icon for any button, please let me know. :slightly_smiling_face:

**Comment by MotoUKRider** [comment]
What a nice addition!

The use of colour, I feel, makes the process much more "fluid" - it allows you to translate an "action" to what is on the map.  For example, when tweaking the Map Layers/Theme, I have to stare quite hard at the buttons to work out just where the action button is - especially when moving between Portrait and Landscape modes.

I'd say that using larger identical icons from the map keeps things "on brand".

**Reply by airald** [reply]
Maybe a different icon with different colour will be more identifiable, so use both options could be a third option.

**Reply by bios4** [reply]
Very nice move, the "old fashioned" icons in the route/map menu made the app somehow look "unfinished".

> so use both options could be a third option.

This sounds nice to me, too!

So switch to the new icons, but keep the color codes for start (green), end (red) and via points (yellow)!?

**Reply by Ciutnik** [reply]
Color is a color, but it would be good to also "standardize" icons within the entire application. We already have two different (not counting the pins on the map) marking the destination of the route. Now comes the checkered flag, which I like as a "finish line" 🙂

<img src="https://github.com/devemux86/cruiser/assets/122647782/5ef381ae-5b86-43e5-ad06-51c21824cac9" width="20%"/>

<img src="https://github.com/devemux86/cruiser/assets/122647782/60925065-c570-4329-a077-89ea6f6e2c29" width="20%"/>

**Reply by devemux86** [reply]
@Ciutnik indeed we can use the same flag icon.

**Reply by Ciutnik** [reply]
> indeed we can use the same flag icon.

Or colorful pins? 😀

**Comment by devemux86** [comment]
Better not add a setting just for these icons. We can choose whether we want same-colored or multi-colored icons.

**Comment by Ciutnik** [comment]
Although I like the top set (single color) more visually, I would vote for the bottom set (colored) so as not to introduce symbols other than those we have on the map. 

Sometimes you have a split second to make a decision, often when you are nervous, it is probably better for everything to be 100% clear and well known at first glance.

**Comment by umpfmpf** [comment]
It looks good like in 2nd row with colour...

**Comment by GuG1050** [comment]
I would prefer the colored proposal.

**Comment by DolomitesTom** [comment]
I vote for one color uniform appearance.

**Reply by Leeb64** [reply]
In terms of visibility, colors would be better.
The appearance is rather secondary...

**Comment by devemux86** [comment]
New icon menu in route planning is available for testing in:
- https://github.com/devemux86/cruiser/discussions/451

It contains the colored waypoint buttons, let's see if you like them.

Please check if everything is working correctly, if we need a different menu order or other icons from [Material icons](https://fonts.google.com/icons?icon.set=Material+Icons&icon.style=Filled).

**Reply by bios4** [reply]
Looks nice, whereas I had preferred the same icon STYLE for all icons, but keep the known color codes for start/end/via.

The only thing I really don't like at all is the icon used for the "more"-tile! I see that it is listed with the "more" tag in the Material symbols, but TO ME it looks like a vertical scroll bar indicator and nothing else...

My suggestion would be to use the "More Horiz" icon (= three horizontal dots) for the tile instead.
Or the "Arrow Right Alt" icon, which would be a nice counterpart of the "back arrow" in all the app menus.

**Reply by devemux86** [reply]
@bios4 thanks for the suggestions. Regular arrow could be confused with the "Extend" button arrows.
Then we need to change the "Extend" icons.

How about one of the following, your "more horiz" icon or "navigate next" icon?

<img src="https://github.com/devemux86/cruiser/assets/3484020/00a50d25-2f65-49df-9d39-fe7c8875a7e5" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/218c8f18-30d0-45dc-b760-49488397e701" width="50%"/>

**Reply by bios4** [reply]
> How about one of the following, your "more horiz" icon or "navigate next" icon?

I vote for "More Horiz" (left) 😁

**Reply by Ciutnik** [reply]
Three dots are a natural "more" link for me.

**Reply by MotoUKRider** [reply]
On first sight, I actually favour the > chevron as it seemed "obvious" - certainly if there's a next page of buttons available.  To me, the ellipsis usually means a truncated portion of text that you can click on to reveal more (like a blog preview).

It's also worth considering that the "more" button may benefit from being the *opposite* of the "back-up" icon when navigating menus.

**Reply by devemux86** [reply]
> It's also worth considering that the "more" button may benefit from being the opposite of the "back-up" icon when navigating menus.

@MotoUKRider This is also correct, but please see the "Extend" button icons with the left / right arrows.

If we use any right arrow at the "More" button and to avoid confusion, then the "Extend" buttons must change icons?

<img src="https://github.com/devemux86/cruiser/assets/3484020/2b51b666-b728-4cdc-95b1-4329aa8fb3e8" width="50%"/>

**Reply by MotoUKRider** [reply]
@devemux86 That's the fun, I guess... You change 1 thing and 3 things then require attention :confused:

**Comment by bios4** [comment]
I have some suggestions to further improve the route planning (menu) - see below & the mockup I created...

1) Move the menu items "Waypoints" and "Turn instructions" from Main menu > Routing to the route options (after long-press route on map); currently it is quite annoying that when shuffling around with way-/via-points I have to leave the route menu and go to the main menu instead to be able to change the waypoints of my route

2) Interchange the tiles for "Delete route" and "Options..." to better match the order of the Main > Routing menu

The "Route diagram" tile which is also available in Main menu > Routing is already available in the route options after selecting the "More..." tile.

<img alt="" src="https://github.com/devemux86/cruiser/assets/145460147/9e9dfbbb-32e3-405c-a9a6-0197dc86e0d5" width="33%">

What do you think @devemux86?

**Reply by devemux86** [reply]
@bios4 thanks for the suggestions!

1. We could probably have "Waypoints" and "Turn instructions" menu at both locations?

2. They are not the same options:
    - Routing menu options opens the active router's options, e.g. BRouter
    - Route menu options opens another route menu (it has ... at the end)

Not sure if the word "Options..." is correct in "Route" menu. We also use it in favorite and waypoint menus.
Perhaps a different icon and word can be used to avoid confusion?

> The "Route diagram" tile which is also available in Main menu > Routing is already available in the route options after selecting the "More..." tile.

This is not the same diagram:
- Routing menu diagram opens the full screen diagram screen
- Route menu diagram turns on / off the diagram at the top

**Reply by bios4** [reply]
> We could probably have "Waypoints" and "Turn instructions" menu at both locations?

Would be OK for me!

> They are not the same options
> Not sure if the word "Options..." is correct in "Route" menu

For me "Options" in both menus are OK, because that's what they are offering: either options for the RoutING, or options for the Route itself. I would stick to both, wording and icon!

My suggestions is more about the general tiles order in the menus, so that the "Options" tile is always before the "Delete" tile...

> Routing menu diagram opens the full screen diagram screen
> Route menu diagram turns on / off the diagram at the top

To be honest, I would expect these two in the exact opposite way!!!

Main > RoutING: turn on/off the diagram in the UI
Route menu: open/show the diagram of the very route I'm currently working with (shuffling around waypoints etc.)

**Reply by devemux86** [reply]
> To be honest, I would expect these two in the absolute opposite way!!!

We can also do this.

**Reply by Leeb64** [reply]
There should be a distinction between both “options”.
Either a different symbol or a different term.
I assumed the same thing between the two "options" and didn't even click on them....

**Comment by JimmyS83** [comment]
Colours always add level of quick recognization. Its harder to design, but if used wiselly, they alway beats one colour icons design.

**Comment by devemux86** [comment]
>  Switching between manual and intelligent routing modes is a bit tricky to achieve.

> We could add the route calculation mode in the route menu, when long press the route line?

<img src="https://github.com/devemux86/cruiser/assets/3484020/878b1b68-4035-4fba-a803-b7aac020a296" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/26bf08ef-ca86-4037-a0a2-acd797e86e23" width="50%"/>

**Reply by bios4** [reply]
This looks very nice!

Regarding a possible German translation: please use "Manuelle Routenberechnung" and "Intelligente Routenberechnung" for the tiles...

For the English naming I would rearrange it also to "Manual route calculation" and "Intelligent route calculation" - this "sounds" a bit more natural to me. 😄

**Reply by devemux86** [reply]
@bios4 the text is the combination of 2 existing texts:
- Route calculation
- Intelligent or Manual

**Reply by bios4** [reply]
That may be the case - it sounds not nice anyway when named like in the screenshot... 😉

In German it would be even wrong when it would be called "Routenberechnung Manuell/Intelligent", because manual/intelligent are not nouns and therefore need to be written with lowercase first letters.

**Reply by devemux86** [reply]
I can add a `:` in between, but prefer to not add new variants for existing texts
especially with so many languages to support. :slightly_smiling_face:

**Reply by bios4** [reply]
See above....

**Reply by devemux86** [reply]
Also the texts are the same as in the Settings.

**Reply by bios4** [reply]
But there they are used as "standalone list items", and not in combination with the word "Routenberechnung" in the same line:

<img alt="" src="https://github.com/devemux86/cruiser/assets/145460147/95f02779-91bf-433c-9878-5ea378e365bb" width="33%">

I just wanted to bring it to your attention, of course you can name it as you see fit...

**Reply by devemux86** [reply]
Here are also separate items:.2 lines with title and value.
I will add a `:` to separate them better visually. :slightly_smiling_face:

**Reply by WolfgangZ42** [reply]
> In German it would be even wrong when it would be called "Routenberechnung Manuell/Intelligent", becaus manual/intelligent are not nouns and therefore need to be written with lowercase first letters.

That's really important for anyone who studied 12 semesters of Germanistik (and now works as Taxi driver ...). 🤭

**Reply by bios4** [reply]
Make fun of it all you want, but I stand for the preservation of the - correct - German language and spelling...

Just take a look at the way people write these days (especially on the internet/social media/text messages). It makes at least ME personally want to vomit.

BTT now

**Reply by MotoUKRider** [reply]
I also found the wording of "Route calculation Intelligent" to be a little odd.
It is interesting to see how non-native speakers write (never mind the txt spk lolz from mobi users on the socialz) :laughing: 

One of my favourites is the phrasing of a question, eg: *"how to move the waypoint to a new place?"*  It make sense, but you know it's a non-native English speaker.  The same with being asked how long you've been here: *"since 10 minutes"* isn't a phrase you'd hear us use.  We also like putting Us in our Favourite Colours!

**Reply by 0709wiwiwi** [reply]
'Allo 'Allo! ?
https://en.wikipedia.org/wiki/%27Allo_%27Allo!

**Reply by Ciutnik** [reply]
> Allo' Allo! 

I've seen it probably 100 times! I absolutely love it*! 😀

*Unfortunately, with the exception of the later casting of Herr Flick and Captain Bertorelli...

@MotoUKRider 
> We also like putting Us in our Favourite Colours!

Since we don't use 'baseball pitch' among the length units and the app is 'made in Europe' maybe we should actually use the forms 'Favourite' and 'Colour' (perhaps there are more such nuances) in the app? 🙂

**Comment by devemux86** [comment]
If the "Options..." and its icon is confusing, we can change it. For a while we also had this:

<img src="https://github.com/devemux86/cruiser/assets/3484020/5425e5eb-7426-4ab3-95fe-16d2ee88bf64" width="50%"/>

**Reply by umpfmpf** [reply]
For 'more' I would perhaps use the 3 dots like everywhere in Android.

**Reply by devemux86** [reply]
@umpfmpf we already use it.

This is for the other menu, now "Options..." on the latest Beta.

**Reply by bios4** [reply]
> If the "Options..." and its icon is confusing, we can change it. 

As I already said above I think it suits very well (way better than the word "Edit" and this gear icon in your latest screenshot).

> Maybe you could rename the options under "Routing" to Settings?

This sounds reasonable, then you could keep the wording "Options" in the route menu.
"Edit" sounds more like it would offer to add/remove/rearrange waypoints of the current route.

**Comment by WolfgangZ42** [comment]
Just a suggestion :

Instead of a 2-level structure, a 3-level chain - okay with some button rearrangements ...

**Status quo:**
![statusquo](https://github.com/devemux86/cruiser/assets/121806238/bb6c0004-5c1d-4a3c-8a7a-0c7d26556191)

**new format:**
![newformat](https://github.com/devemux86/cruiser/assets/121806238/41f83ec0-13f3-4691-9fbe-d171e5d44f15)

Open fire for discussions!

**Reply by devemux86** [reply]
@WolfgangZ42 thanks, this is a nice idea with one less button in the first screen.

**Reply by devemux86** [reply]
Another idea is to combine the first 2 screens into 1 screen with all the buttons related to the route.

But it will be long and use a scrollbar, so probably not very useful vs. 2 screens with a More button?

**Reply by bios4** [reply]
> But it will be long and use a scrollbar, so probably not very useful vs. 2 screens with a More button?

Please DON'T implement a menu with scrollbars!!!

**Comment by devemux86** [comment, marked as answer]
New icon menu in route planning is available in [Cruiser 3.0.41](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
